feat(text-input): text-input @carbon/react v11 sync (#10308)
carbon-design-system/carbon-for-ibm-dotcom: packages/carbon-web-components/docs/form-story.mdx

# Having components participate in form

> Though form elements in `carbon-web-components` (e.g. `<cds-text-input>`) are not native form elements like `<input>`, form elements in `carbon-web-components` have some extra APIs that align well to web/framework standards that allow those form elements to participate in form.

Reviewer: ```suggestion
Though form elements in `@carbon/web-components` (e.g. `<cds-text-input>`) are not native form elements like `<input>`, they have some extra APIs that align well with web/framework standards.
```

  });
  event.formData = formData;
  form.dispatchEvent(event);

>   // Now `formData` is populated with the data in components such as `<cds-text-input>` or `<cds-dropdown>`, etc. in the `<form>`.

Reviewer: ```suggestion
  // `formData` is populated with the data in components such as `<cds-text-input>` or `<cds-dropdown>`, etc. in the `<form>`.
```
